hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -240,22 +244,21 @@ protected void finishRollback(HoodieInstant inflightInstant, HoodieRollbackMetad
         this.txnManager.beginTransaction(Option.of(inflightInstant), Option.empty());
       }
 
-      // If publish the rollback to the timeline, we first write the rollback metadata
-      // to metadata table
+      // If publish the rollback to the timeline, we first write the rollback metadata to metadata table
+      // Then transition the inflight rollback to completed state.
       if (!skipTimelinePublish) {
         writeTableMetadata(rollbackMetadata);
-      }
-
-      // Then we delete the inflight instant in the data table timeline if enabled
-      deleteInflightAndRequestedInstant(deleteInstants, table.getActiveTimeline(), resolvedInstant);
-
-      // If publish the rollback to the timeline, we finally transition the inflight rollback
-      // to complete in the data table timeline
-      if (!skipTimelinePublish) {
         table.getActiveTimeline().transitionRollbackInflightToComplete(inflightInstant,
             TimelineMetadataUtils.serializeRollbackMetadata(rollbackMetadata));
         LOG.info("Rollback of Commits " + rollbackMetadata.getCommitsRollback() + " is complete");
       }
+
+      // Commit to rollback instant files are deleted after the rollback commit is transitioned from inflight to completed
+      // If job were to fail after transitioning rollback from inflight to complete and before delete the instant files,
+      // then subsequent retries of the rollback for this instant will see if there is a completed rollback present for this instant
+      // and then directly delete the files and abort.
```

Comment:
can you point me to this part of the code. Where we check for completed rollback instants and clean up the files directly ? 
From what I know, we only check for pending rollbacks 
ref: getPendingRollbackInfos
and re-use the rollback instant if applicable. 

Also, do we have a test for this ?

Reply:
I committed the cleanup logic. Also, added test for this change.